daemon/volumes.go

```diff
@@ -163,11 +165,29 @@ func (container *Container) parseVolumeMountConfig() (map[string]*Mount, error)
 		if m, exists := mounts[mountToPath]; exists {
 			return nil, fmt.Errorf("Duplicate volume %q: %q already in use, mounted from %q", path, mountToPath, m.volume.Path)
 		}
+
+		runtime.LockOSThread()
```

Comment:
Yes SELinux labeling sets up a per process/thread label,  This is basically saying if this thread creates any content on disk label it with mountlabel.

I did not know UnlockOSThread needed to be called, but no problem adding it.